File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Text/updateshapetext.rb
module Asposediagramjava
  module UpdateShapeText
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        shapes = diagram.getPages().getPage(0).getShapes()
        
        i = 0
        while i < shapes.getCount()
            shape = shapes.get(i)
            if shape.getNameU() == "Process" && shape.getID() == 1
                shape.getText().getValue().clear()
                shape.getText().getValue().add(Rjb::import('com.aspose.diagram.Txt').new("New Text"))
            end
            i +=1
        end
        
        # Save diagram
        diagram.save(data_dir + "UpdateShapeText.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Updated shape text."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithText/UpdateShapeText.php
<?php
namespace Aspose\Diagram\WorkingwithText;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\Txt as Txt;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class UpdateShapeText{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram=new Diagram($dataDir."Drawing.vsd");

        $shapes=$diagram->getPages()->getPage(0)->getShapes();
        $i=0;
        while($i<(int)(string)$shapes->getCount()) {
        $shape = $shapes->get($i);
        if ($shape->getNameU() == "Process" && (int)(string)$shape->getID() == 1) {
        $shape->getText()->getValue()->clear();
        $shape->getText()->getValue()->add(new Txt("New Text"));
        }
        $i += 1;
        }

        # Save diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."UpdateShapeText.vdx",$saveFileFormat->VDX);
        print "Updated shape text.".PHP_EOL;
    }
}